# Analysis of gut microbiota in Qiime2 and Rstudio/R.

## Workflow overview and output
![image](https://github.com/user-attachments/assets/406d3dd4-50f6-4f06-85dd-5a6b694dd4c3)

## [Quality check of raw reads](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Quality_Check.md)

## [Step in Qiime 2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime2_steps.md)
### [1. Importing raw data into Qiime2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/1.%20Importing-raw-data-into-Qiime2.md)

### [2. Filtering, dereplication, sample inference, chimera identification, and merging of paired-end reads by DADA2 package in qiime2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/2.%20Filtering%2C%20dereplication%2C%20sample%20inference%2C%20chimera%20identification%2C%20and%20merging%20of%20paired-end%20reads%20by%20DADA2%20package%20in%20qiime2.md)

### [3. Training a primer-based region-specific classifier for taxonomic classification by Na√Øve-Bayes method](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/3.%20Training%20a%20primer-based%20region-specific%20classifier%20for%20taxonomic%20classification%20by%20Na%C3%AFve-Bayes%20method%20(in%20Qiime2).md)

### [4. Creating a phylogenetic tree using align-to-tree-MAFFT-FastTree](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/4.%20Creating%20a%20phylogenetic%20tree%20using%20align-to-tree-MAFFT-FastTree.md)

### [5.  Rarefaction Analysis and Core Diversity Metrics Calculation](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/5.%20Rarefraction%20curve%20analysis.md)

### [6. Alpha Diversity analysis](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/6.%20Alpha%20Diversity%20analysis.md) 

### [7. Beta Diversity analysis](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/7.%20Beta%20Diversity%20analysis.md)

### [8. Export file from Qiime2 for steps in R/Rstudio](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/8.%20Export%20file%20from%20Qimme2%20for%20R%20steps.md)


## [Steps in R/Rstudio](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/R_steps/R_step.md)
### [1. Differential abundance analysis of taxa using ANCOMBC](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/R_steps/1.%20Differential%20abundance%20analysis%20of%20taxa%20by%20ANCOMBC.md)

### [2. Analysis of biomarkers for gut microbiome](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/R_steps/2.%20Analysis%20of%20biomarkers%20for%20gut%20microbiome.md)
