# Analysis of gut microbiota in Qiime2 and Rstudio/R.

## Workflow overview and output
(Insert the workflow)

## [Quality check of raw reads](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Quality_Check.md)

## [Step in Qiime 2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime2_steps.md)
### [1. Importing raw data into Qiime2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/1.%20Importing-raw-data-into-Qiime2.md)

### [2. Filtering, dereplication, sample inference, chimera identification, and merging of paired-end reads by DADA2 package in qiime2](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/2.%20Filtering%2C%20dereplication%2C%20sample%20inference%2C%20chimera%20identification%2C%20and%20merging%20of%20paired-end%20reads%20by%20DADA2%20package%20in%20qiime2.md)

### 3. Training a primer-based region-specific classifier for taxonomic classification by Na√Øve-Bayes method

### 4. Creating a phylogenetic tree using align-to-tree-MAFFT-FastTree

### 5.  Rarefaction Analysis

### 6. Core Diversity Metrics Calculation

### 7. Alpha Diversity analysis 

### 8. Beta Diversity analysis

## Steps in R/Rstudio
### 1. Differential abundance analysis of taxa by ANCOMBC

### 2. Analysis of chemical biomarkers for gut microbiome: Heatmap correlation
