## Chao1
Chao1 alpha diversity analysis estimates species richness in a sample. It focuses on rare ASVs (appearing only once or twice), helping to assess how many species might be undetected due to limited sequencing depth. This is useful for comparing microbial richness across sample groups.
```bash
qiime diversity alpha \
  --i-table filtered-table.qza \
  --p-metric chao1 \
  --o-alpha-diversity chao1-diversity.qza
```
```bash
qiime metadata tabulate \
  --m-input-file chao1-diversity.qza \
  --o-visualization chao1-diversity.qzv
```
![image](https://github.com/user-attachments/assets/41a865d0-0e3e-4cf3-badb-3b6a511bfa0d)

```bash
qiime diversity alpha-group-significance \
  --i-alpha-diversity chao1-diversity.qza \
  --m-metadata-file metadata.tsv \
  --o-visualization chao1-group-significance.qzv
```
[chao1-group-significance.qzv](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/chao1-diversity-significance.qzv) is generated.
![image](https://github.com/user-attachments/assets/420ce7d1-21bd-45b6-9258-c629552e0223)
The Kruskal-Wallis test results indicate:
    *Overall comparison (all groups)*
        H = 7.43, p-value = 0.115 → No significant difference in Chao1 alpha diversity among the groups (p > 0.05).
    *Pairwise comparisons*
        Most group comparisons have p-values > 0.05, suggesting no statistically significant differences.
        However, RP-20 vs RP-40 and RP-20 vs RP-5 have p-values = 0.0495, indicating a potential difference.
        Yet, the q-values are > 0.05 (after multiple testing correction), meaning the observed differences may not be strong enough to be considered statistically significant.    
**Conclusion**: There is no significant difference in Chao1 alpha diversity among the groups after multiple comparison correction.

## Shannon index
The Shannon index measures alpha diversity, accounting for both species richness (number of species) and evenness (distribution of species abundances).
    Higher Shannon index → More diverse and evenly distributed microbial community.
    Lower Shannon index → A community dominated by a few species, indicating lower diversity.
```bash
qiime diversity alpha-group-significance \
  --i-alpha-diversity shannon-diversity.qza \
  --m-metadata-file metadata.tsv \
  --o-visualization shannon-group-significance.qzv
```
[shannon-group-significance.qzv](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/shannon-group-significance.qzv) file is generated.
![image](https://github.com/user-attachments/assets/16ed9d07-b543-4eb6-8c06-f32130231ffc)
Overall: p-value = 0.6476 → There is no significant difference in the Shannon index between groups overall. This means that the microbial diversity structure across all groups is similar.
Pairwise: All p-values > 0.05 → There is no significant difference in the Shannon index between any pair of groups. This indicates that no group has significantly higher or lower microbial diversity compared to the others.    
**Conclusion: The groups have equivalent microbial diversity, suggesting that the grouping factor (e.g., experimental condition) does not strongly influence gut microbiome diversity in this dataset**
## Pielou's Evenness Index
Pielou's Evenness Index measures the evenness of species distribution in a community. It indicates how evenly the species are distributed, with values ranging from 0 (completely uneven) to 1 (completely even).

```bash
qiime metadata tabulate \
  --m-input-file evenness_vector.qza \
  --m-metadata-file metadata.tsv \
  --o-visualization evenness_boxplot.qzv
```
[evenness_boxplot.qzv](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/evenness-boxplot.qzv) file is generated
![image](https://github.com/user-attachments/assets/a0d29b65-06ec-4551-8f86-258f9d953cd3)
Overall: H = 6.78, p-value = 0.148 → No significant difference in evenness between groups.
Pairwise: All pairwise comparisons have p-value > 0.05, indicating no significant differences in evenness between any pair of groups.    
**Conclusion:There is no significant difference in Pielou's Evenness Index across groups, suggesting that the evenness of microbial distribution is similar in all groups**

## Faith's Phylogenetic Diversity (Faith's PD) 
Faith's PD measures the total branch length of a phylogenetic tree that connects all species in a sample. It reflects both species richness and phylogenetic diversity, considering evolutionary relationships.
A higher Faith’s PD indicates a more diverse microbial community with greater evolutionary variety, while a lower Faith’s PD suggests a more phylogenetically constrained community.
```bash
qiime diversity alpha-group-significance \
  --i-alpha-diversity faith-pd.qza \
  --m-metadata-file metadata.tsv \
  --o-visualization faith-pd-group-significance.qzv
```
[faith-pd-group-significance.qzv](https://github.com/thaocaoHPzbook/Goldfish-16S-rRNA-amplicon-data-analysis/blob/main/Qiime_steps/faith_pd_group_significance.qzv) is generated
![image](https://github.com/user-attachments/assets/ce48b47b-18c9-4934-b5f3-a40cc7e0bcdc)
Overall (All groups): H-statistic = 1.33, p-value = 0.8563 → No significant difference in Faith's PD between the groups.    
Pairwise (Between groups):All pairwise comparisons have p-value > 0.05, indicating no significant differences in Faith's PD between any of the groups.    
**Conclusion: There is no significant difference in Faith's PD between the groups, suggesting that the phylogenetic diversity is similar across all groups in this study.**
